hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1750,10 +1751,16 @@ public boolean skipRegionManagementAction(final String action) {
     return false;
   }
 
-  public boolean balance(boolean force) throws IOException {
-    if (loadBalancerTracker == null || !loadBalancerTracker.isBalancerOn()) {
+  public boolean balance(BalanceRequest request) throws IOException {
+    return balance(request, false);
+  }
+
+  private boolean balance(BalanceRequest request, boolean isChore) throws IOException {
```

Comment:
What's this new business about `isChore`?

Reply:
I figured we might want to differentiate between requests coming from the user and those from the ongoing chore. I'll remove it.

Reply:
That's a reasonable thing for an operator/admin to want to know. There's a similar concept in the Normalizer in the form of an `isHighPriority` flag, which is true for a user request and false for a chore-generated request. It doesn't look like the distinction is logged.

I assume this is something you've wished you'd had in production. If so, I think we should keep it.

Reply:
Actually to be honest this was just an artifact of some interface evolutions that have occurred on this PR. I'm fine leaving it out. I do think sometimes it would be helpful to differentiate between the chore and manual, but I'm not sure this line would be the most important or only one. I'll leave this out for now.